From 2b8f46ed079f54cecfd29115d3be0296da2738a4 Mon Sep 17 00:00:00 2001
From: Benjamin Peterson <benjamin@engflow.com>
Date: Mon, 31 Jul 2023 08:04:21 -0700
Subject: [PATCH 7/7] bazel: Get rid of exec_tools. (#13401)

Bazel has removed this attribute in bazelbuild/bazel@c061e57a7004a88eeb2f84d094d9a88b56c146b6.

Closes #13401

COPYBARA_INTEGRATE_REVIEW=https://github.com/protocolbuffers/protobuf/pull/13401 from benjaminp:exec_tools 9e718ff0fd11ff1fe70ed3e2536373792504c9c5
PiperOrigin-RevId: 552482730

Backport of commit 1bee4578b8a812faed90432798222670f6df2a9b
---
 build_defs/internal_shell.bzl   | 4 ++--
 objectivec/BUILD.bazel          | 2 +-
 src/google/protobuf/BUILD.bazel | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/build_defs/internal_shell.bzl b/build_defs/internal_shell.bzl
index 649184a51..91628a5e5 100644
--- a/build_defs/internal_shell.bzl
+++ b/build_defs/internal_shell.bzl
@@ -32,7 +32,7 @@ def inline_sh_binary(
     native.genrule(
         name = name + "_genrule",
         srcs = srcs,
-        exec_tools = tools,
+        tools = tools,
         outs = [name + ".sh"],
         cmd = "cat <<'EOF' >$(OUTS)\n#!/bin/bash -exu\n%s\nEOF\n" % cmd,
         visibility = ["//visibility:private"],
@@ -77,7 +77,7 @@ def inline_sh_test(
     native.genrule(
         name = name + "_genrule",
         srcs = srcs,
-        exec_tools = tools,
+        tools = tools,
         outs = [name + ".sh"],
         cmd = "cat <<'EOF' >$(OUTS)\n#!/bin/bash -exu\n%s\nEOF\n" % cmd,
         visibility = ["//visibility:private"],
diff --git a/objectivec/BUILD.bazel b/objectivec/BUILD.bazel
index 107983806..c59bf0e60 100644
--- a/objectivec/BUILD.bazel
+++ b/objectivec/BUILD.bazel
@@ -42,7 +42,7 @@ genrule(
         for wkt in _OBJC_WKT_NAMES
         for ext in _OBJC_EXTS
     ]),
-    exec_tools = ["//:protoc"],
+    tools = ["//:protoc"],
     tags = ["manual"],
 )
 
diff --git a/src/google/protobuf/BUILD.bazel b/src/google/protobuf/BUILD.bazel
index f7f1c7b48..735610170 100644
--- a/src/google/protobuf/BUILD.bazel
+++ b/src/google/protobuf/BUILD.bazel
@@ -138,7 +138,7 @@ genrule(
             --proto_path=$$(dirname $$(dirname $$(dirname $(location any.proto)))) \
             $(SRCS)
     """,
-    exec_tools = ["//:protoc"],
+    tools = ["//:protoc"],
     visibility = ["//visibility:private"],
 )
 
-- 
2.42.1

